automation/TaskManager.py

```diff
@@ -243,20 +243,20 @@ def _manager_watchdog(self) -> None:
             # 300 second buffer to avoid killing freshly launched browsers
             # TODO This buffer should correspond to the maximum spawn timeout
             if self.process_watchdog:
-                browser_pids: Set[int] = set()
+                geckodriver_pids: Set[int] = set()
                 display_pids: Set[int] = set()
                 check_time = time.time()
                 for browser in self.browsers:
-                    if browser.browser_pid is not None:
-                        browser_pids.add(browser.browser_pid)
+                    if browser.geckodriver_pid is not None:
+                        geckodriver_pids.add(browser.geckodriver_pid)
                     if browser.display_pid is not None:
                         display_pids.add(browser.display_pid)
                 for process in psutil.process_iter():
                     if process.create_time() + 300 < check_time and (
                             (process.name() == 'firefox' and (
```

Comment:
@englehardt seeing as this is the geckodriver process I don't think it'll ever be named `firefox`
```suggestion
                            (process.name() == 'geckodriver' and (
```

Reply:
@vringar There definitely could be processes named Firefox that aren't closed properly. I'm wondering whether we should add an additional check for geckodriver processes instead. Note that the geckodriver process spawns the firefox process, so we may want to do something like [this](https://github.com/mozilla/OpenWPM/blob/e42e79fcb3eb2cc6d08030208c66ccab3e3c0c2a/automation/BrowserManager.py#L367-L384).